app/models/menu_item.rb

```diff
@@ -0,0 +1,43 @@
+class MenuItem
```

Comment:
what a wonderful example of what I am talking about [here](https://community.openfoodnetwork.org/t/how-to-name-a-new-service-in-ofn/1530/7?u=luisramos0). option 6, non-ORM data objects like this should be in /services. It's funny how this is very obvious to my large scale java/c# service oriented conditioned brain but goes against MVC and "normal Rails". I think we should still follow it :-)
Basically, I advocate that /models is for ORM only.

Reply:
Spree is a very good example of a mess of ORM and logic mixed together in /models, with for example shipments (db entity) mixed with packages (service entity).

Reply:
It's not that obvious to me. It represents the data of a menu item. It doesn't fetch data from the database but from the routes, the locale and the environment. You could separate the logic into a stupid MenuItem, a pure data structure, and a MenuItemFactory which is a service creating those items. But having it in one place is nice as well. It's not a big class and follows the OO pattern of encapsulation.

I think it shouldn't matter where data is stored. Not every model has to be an ActiveRecord object.

Reply:
I dont think we need to split between service and the business model. The point is that MenuItem is a business model with business logic (the mentioned option 6) and belongs in the service layer, never in the data access layer.

"I think it shouldn't matter where data is stored."
This is exactly where I disagree. I think this it the open door to the big mess we (and spree) have under /models. And if we dont start to move things to services (and organize services well) we will keep having a bigger mess under /models. It wont stop growing and it's already over 100 files.

Reply:
A well organized app will look simple (very different from what we have). It will have only ActiveRecord  objects under /models and then in /services you will find a well separated list of service modules under different folders, each addressing specific parts of the business logic. This is where the complexity of the app will be gracefully addressed far from active record relations and callbacks. I have seen this in many apps and it makes a huge difference as the app grows in complexity.

We have some serious design errors in spree/ofn, for example, the logic to address adjustments (or the logic in order updater) has been put into /models and it should be 100% outside of it under /services and completely detached from callbacks and AR relations.

Reply:
so my recommendation is to put this file as it is under engines/web/app/services and not together with the other +100 files under /models

Reply:
Ok, my 2 cents here.

IMO this is a view-related abstraction and at first, I wanted to add it to `app/views` or perhaps in a new `app/presenters/` then I thought, "they won't like it and what I want to be addressed quickly will take longer. I don't care about the actual folder but the code changeability. Change as less a possible and hope for the best".

Then, if you ask me, as for my experience and the code conventions I came to use over time, aligned with the Rails community (I think), `models/` does not only contain AR objects but domain-related models not backed by a DB table. This grows over time as the team and the app mature. `services/` is used for command-like, purpose-specific classes; they perform an action. Things get larger and that is when folders like `queries/` and `presenters/` come up. They all benefit from the Rails autoloading: any Ruby code within `app/*/` will get evaluated and ready to be used.

I was quite strict about these things when I was younger but I came to realize that because of autoloading it doesn't matter much where you put things (and be happier :joy: ). It is the right OO abstraction and the overall code quality what makes an app changeable. It might even matter less for me as I use fuzzy finder and ctags. 

I also have the feeling that I waste my time in these discussions I want to rather focus on providing value. So, all the explanation is to conclude that I'm absolutely happy with anything you guys decide. I'm fine with any arbitrary decision we come up with as a team :muscle: .

As a side note, however, the mess with AR callbacks and logic you mention @luisramos0 is something all Rails app tend to suffer after sometime due to the way Rails couples it all in ActiveModel objects. It makes it really easy at first but requires OO discipline to overcome it when your app becomes something else than an MVP. Discipline teams need to learn, which takes time. IMO where you put the mess won't change things, either `models/`, `services/` or anywhere else.

Reply:
So, to keep things agile, I'll go and move it to `services/` as that is what we've been doing lately.

Reply:
I agree, let's not waste our time in long discussions, I raised this point because I think this topic is very important. imo MVC thinking is poison at this stage and we need to focus on service oriented design separated from the data access layer.